File: 0179-largest-number/0179-largest-number.java
class Solution {
    public String largestNumber(int[] nums) {

        int n = nums.length;

        String[] str = new String[n];

        for(int i=0;i< n;i++){
            str[i] = String.valueOf(nums[i]);
        }

        Arrays.sort(str, (a,b) -> (b+a).compareTo(a+b));

        StringBuilder sb  = new StringBuilder();

        for(String s : str){
            sb.append(s);
        }

        String result = sb.toString();

        return result.startsWith("0") ? "0" : result;
        
    }
}

File: 0179-largest-number/0179-largest-number.js
/**
 * @param {number[]} nums
 * @return {string}
 */
var largestNumber = function(nums) {

    let strs = nums.map(String);

    strs.sort((a, b) => {
        return (b+a).localeCompare(a+b);
    })

    let res = strs.join("");

    if(res[0] === "0") return "0";

    return res;
    
};